discovery-provider/nginx_conf/main.lua

```diff
@@ -13,6 +13,54 @@ math.randomseed(string.byte(resty_random.bytes(1)))
 
 local _M = {}
 
+local redirect_target_weights = {}
+for index, redirect_target in ipairs(config.redirect_targets) do
+    redirect_target_weights[redirect_target] = 1
+end
+
+function get_request_counts (discovery_providers)
+    local httpc = resty_http.new()
+    request_counts = {}
+
+    for index, discovery_provider in ipairs(discovery_providers) do
+        local res, err = httpc:request_uri(discovery_provider .. "/health_check", { method = "GET" })
+
+        if not res then
+            ngx.log(ngx.ERR, "failed to get request count from discovery provider: ", discovery_provider, err)
+            request_count[discovery_provider] = -1
+        else
+            request_count[discovery_provider] = tonumber(res.body)
+        end
+    end
+
+    return request_counts
+end
+
+function update_redirect_target_weights (premature)
+    if premature then
+        return
+    end
+
+    local request_count = get_request_counts(config.redirect_targets)
```

Comment:
we should read redirect target from python and remove this env var entire (or keep as an override)